lntemp/harness.go

```diff
@@ -80,3 +93,376 @@ func NewHarnessTest(t *testing.T, lndBinary string,
 		lndErrorChan: make(chan error, 10),
 	}
 }
+
+// Start will assemble the chain backend and the miner for the HarnessTest. It
+// also starts the fee service and watches lnd process error.
+func (h *HarnessTest) Start(chain node.BackendConfig, miner *HarnessMiner) {
+	// Spawn a new goroutine to watch for any fatal errors that any of the
+	// running lnd processes encounter. If an error occurs, then the test
+	// case should naturally as a result and we log the server error here
+	// to help debug.
+	go func() {
+		select {
+		case err, more := <-h.lndErrorChan:
+			if !more {
+				return
+			}
+			h.Logf("lnd finished with error (stderr):\n%v", err)
+
+		case <-h.runCtx.Done():
+			return
+		}
+	}()
+
+	// Start the fee service.
+	h.feeService = startFeeService(h.T)
+
+	// Assemble the node manager with chainBackend and feeServiceURL.
+	h.manager.chainBackend = chain
+	h.manager.feeServiceURL = h.feeService.url
+
+	// Assemble the miner.
+	h.Miner = miner
+}
+
+// ChainBackendName returns the chain backend name used in the test.
+func (h *HarnessTest) ChainBackendName() string {
+	return h.manager.chainBackend.Name()
+}
+
+// SetUp starts the initial seeder nodes within the test harness. The initial
+// node's wallets will be funded wallets with 10x10 BTC outputs each.
+func (h *HarnessTest) SetupStandbyNodes() {
+	h.Log("Setting up standby nodes Alice and Bob...")
+	defer h.Log("Finshed the setup, now running tests...")
+
+	lndArgs := []string{
+		"--default-remote-max-htlcs=483",
+		"--dust-threshold=5000000",
+	}
+	// Start the initial seeder nodes within the test network, then connect
+	// their respective RPC clients.
+	h.Alice = h.NewNode("Alice", lndArgs)
+	h.Bob = h.NewNode("Bob", lndArgs)
+
+	// First, make a connection between the two nodes. This will wait until
+	// both nodes are fully started since the Connect RPC is guarded behind
+	// the server.Started() flag that waits for all subsystems to be ready.
+	h.ConnectNodes(h.Alice, h.Bob)
+
+	addrReq := &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
```

Comment:
I wonder if we should either change this to be a taproot addr, or give them a 50/50 mix of taproot and p2wkh addresses. This might help shake out some lingering issues w/ the taproot default integration.

Reply:
I think so. Maybe we could make it configurable for the itest, and all the default addresses will change accordingly, which can be done in a follow up PR, just to keep the current PR small.